### ISSUE TITLE
Selectable light and dark editor themes for Follow System Theme feature

### ISSUE THREAD

**Issue by jefemagril** [godotengine/godot-proposals#10578]
### Describe the project you are working on

I am using the Godot editor to make games.

### Describe the problem or limitation you are having in your project

I like the macOS feature of automatically switching the system color theme between dark and light mode based on the time of day. Windows can do this too using a third-party app "Windows Auto Dark Mode". I found the editor setting under Interface->Theme: "Follow System Theme". However, it works only partially and very poorly. For some reason, this feature takes the choice of specific color scheme out of your control and chooses a very bad one for both light and dark.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Right now the editor settings show a Preset dropdown under the Follow System Theme tickbox. I propose that when Follow System Theme is selected, the Preset line is replaced with two lines of dropdowns: Light Preset and Dark Preset. Then, Godot will simply apply the selected presets corresponding to the system theme.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I haven't looked at the code yet, and I can, but I imagine this would be a very simple enhancement to apply. Mostly just reusing existing code.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Users can manually change the editor color theme. A @tool script might be difficult because it would need to read the system theme.

### Is there a reason why this should be core and not an add-on in the asset library?

Because the feature already exists Follow System Theme but it works so poorly I doubt anyone is using it.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/2938 (which is about script editor themes).

**Comment by HaipaDev** [comment]
Really looking forward to this as I also have started using time based automatic theme switching
Light theme during the day is superior just like dark mode is made for the dark, and we should be able to select custom themes for each

**Comment by Joseph-DiGiovanni** [comment]
> I propose that when Follow System Theme is selected, the Preset line is replaced with two lines of dropdowns

IIRC this is not possible as there is no way to conditionally show options.

I think we could get away with a dropdown for light and dark presets, but a custom preset is not feasible without cluttering the settings.

**Comment by norrbru** [comment]
> > I propose that when Follow System Theme is selected, the Preset line is replaced with two lines of dropdowns
> 
> IIRC this is not possible as there is no way to conditionally show options.
> 
> I think we could get away with an extra set of dropdowns for light and dark presets, but a custom preset is not feasible without cluttering the settings.

I don't think it's necessary to conditionally show or hide anything. For example, most IDEs handle this by just having two lines of dropdowns always visible, one for the `Light Theme` and one for the `Dark Theme`. 

I don't think this would cause clutter at all, and don't think it's a problem to have them always be visible. If anything, it effectively communicates to the user that there is an option to easily switch themes with the system, for example.

**Comment by Joseph-DiGiovanni** [comment]
They were referring to the duplicated theme options for both light and dark custom themes, not just the theme selection options. Each would need a background color, accent color, text colors, etc to be complete.

Maybe it could get its own page as to not clutter the primary theme settings if it were to be implemented.

**Comment by yak3d** [comment]
I agree that the original feature feels hindered by not letting me choose color schemes, to the point where I just don't engage with it.

I like to use Dracula during the night, and its light theme Alucard during the day. If I can't specify those themes for day vs. night I just won't use this feature.
